File: backend/Models/Category.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Vietsov.Api.Models;

public enum CategoryType
{
    Event,
    NewsType,
    Other
}

public class Category
{
    [Key]
    public int Id { get; set; }

    [Required]
    [MaxLength(200)]
    public string Name { get; set; } = string.Empty;

    [Required]
    [MaxLength(200)]
    public string Slug { get; set; } = string.Empty;

    [Required]
    public CategoryType Type { get; set; } = CategoryType.Other;

    [Column(TypeName = "text")]
    public string? Description { get; set; }

    public int? ParentId { get; set; }

    [ForeignKey("ParentId")]
    public virtual Category? Parent { get; set; }

    public virtual ICollection<Category> Children { get; set; } = new List<Category>();

    public bool IsActive { get; set; } = true;

    public int Order { get; set; } = 0;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    public virtual ICollection<ArticleCategory> ArticleCategories { get; set; } = new List<ArticleCategory>();
}

File: frontend/src/types/category.ts
export enum CategoryType {
  EVENT = "event",
  NEWS_TYPE = "news_type",
  OTHER = "other",
}

export interface Category {
  id: number;
  name: string;
  slug: string;
  type: CategoryType;
  parentId?: number;
  parent?: Category;
  children?: Category[];
  description?: string;
  isActive: boolean;
  order: number;
  createdAt: string;
  updatedAt: string;
}

